src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java

```diff
@@ -124,8 +124,9 @@ protected void writeTo(OutputStream os) throws IOException {
     }
 
     protected void writeTo(OutputStream os, @NonNull ContentFilter filter) throws IOException {
-        if (isBinary) {
+        if (isBinary || filter == ContentFilter.NONE) {
             writeTo(os);
+            return;
```

Comment:
Drop the `return` and `preservesLineEndingsWithContentFilterNone` starts failing because the content is written twice.